src/main/java/org/opensrp/web/rest/ClientResource.java

```diff
@@ -100,4 +155,110 @@ public Client update(Client entity) {//TODO check if send property and id matche
+		searchBean.setClientType(clientType);
+		searchBean.setProviderId(getStringFilter(PROVIDERID, request));
+		
+		if (pageNumberParam != null) {
+			pageNumber = Integer.parseInt(pageNumberParam) - 1;
+		}
+		if (pageSizeParam != null) {
+			pageSize = Integer.parseInt(pageSizeParam);
+		}
+		
+		searchBean.setPageNumber(pageNumber);
+		searchBean.setPageSize(pageSize);
+		AddressSearchBean addressSearchBean = new AddressSearchBean();
+		
+		String attributes = getStringFilter("attribute", request);
+		searchBean.setAttributeType(StringUtils.isEmptyOrWhitespaceOnly(attributes) ? null : attributes.split(":", -1)[0]);
+		searchBean.setAttributeValue(StringUtils.isEmptyOrWhitespaceOnly(attributes) ? null : attributes.split(":", -1)[1]);
+		
+		if (clientType.equalsIgnoreCase(HOUSEHOLD)) {
+			return getHouseholds(searchBean, addressSearchBean);
+		} else if (clientType.equalsIgnoreCase(HOUSEHOLDMEMEBR)) {
+			
+			clientList = clientService.findMembersByRelationshipId(baseEntityId);
+			clientsArray = (JsonArray) gson.toJsonTree(clientList, new TypeToken<List<Client>>() {}.getType());
+			response.put("clients", clientsArray);
+			
+		} else if (clientType.equalsIgnoreCase(ALLCLIENTS)) {
+			searchBean.setClientType(HOUSEHOLD);
+			return getAllClients(searchBean, addressSearchBean);
+		}
+		
+		return new ResponseEntity<>(new Gson().toJson(response), HttpStatus.OK);
+		
+	}
+	
+	public ResponseEntity<String> getAllClients(ClientSearchBean clientSearchBean, AddressSearchBean addressSearchBean) {
+		
+		Map<String, Object> response = new HashMap<String, Object>();
+		JsonArray clientsArray = new JsonArray();
+		
+		List<Client> clients = clientService.findAllClientsByCriteria(clientSearchBean, addressSearchBean);
+		
+		total = getTotal(clientSearchBean, addressSearchBean);
+		
+		clientsArray = (JsonArray) gson.toJsonTree(clients, new TypeToken<List<Client>>() {}.getType());
+		response.put("clients", clientsArray);
+		response.put("total", total);
+		return new ResponseEntity<>(new Gson().toJson(response), HttpStatus.OK);
+	}
+	
+	public ResponseEntity<String> getHouseholds(ClientSearchBean clientSearchBean, AddressSearchBean addressSearchBean) {
+		
+		DateTime[] lastEdit = null;
+		Map<String, Object> response = new HashMap<String, Object>();
+		JsonArray clientsArray = new JsonArray();
+		List<Client> clients = new ArrayList<Client>();
+		List<String> ids = new ArrayList<String>();
+		List<Client> _clients = new ArrayList<Client>();
+		clients = clientService.findHouseholdByCriteria(clientSearchBean, addressSearchBean, lastEdit == null ? null
+		        : lastEdit[0], lastEdit == null ? null : lastEdit[1]);
+		if (clients.size() != 0) {
+			for (Client client : clients) {
+				ids.add(client.getBaseEntityId());
+			}
+			total = getTotal(clientSearchBean, addressSearchBean);
+			_clients = clientService.getHouseholdList(ids, clientSearchBean.getClientType(), addressSearchBean,

```

Comment:
Is there a way to avoid 2 queries here? 

Perhaps write one method in clientService like **_clientService.getHouseholdListByCriteria_** which will at the database level invoke only one custom sql query on the mapper

Reply:
I think that will possible.

Reply:
Now only one query run to databse